lib/flume/pipeline/control/options.ex

```diff
@@ -0,0 +1,48 @@
+defmodule Flume.Pipeline.Control.Options do
+  @option_specs [
+    async: [type: :boolean, default: false],
+    temporary: [type: :boolean, default: true]
+  ]
+
+  @doc """
+  Returns options for pausing/resuming a pipeline
+  ### Examples
+      iex> Flume.Pipeline.Control.Options.sanitized_options
+      {:ok, [temporary: true, async: false]}
+      iex> Flume.Pipeline.Control.Options.sanitized_options([unwanted: "option", async: false])
+      {:ok, [temporary: true, async: false]}
+      iex> Flume.Pipeline.Control.Options.sanitized_options([temporary: true, async: true, extra: "value"])
+      {:ok, [temporary: true, async: true]}
+      iex> Flume.Pipeline.Control.Options.sanitized_options([temporary: 1, async: false])
+      {:error, "expected :temporary to be a boolean, got: 1"}
+      iex> Flume.Pipeline.Control.Options.sanitized_options([temporary: false, async: 0])
+      {:error, "expected :async to be a boolean, got: 0"}
+  """
+  @type option_spec :: [
+          async: boolean(),
+          temporary: boolean()
+        ]
+
+  @spec sanitized_options(option_spec()) :: {:ok, option_spec()} | {:error, String.t()}
+  def sanitized_options(options \\ []) do
+    Enum.reduce_while(@option_specs, {:ok, []}, fn {key, _} = ele, {:ok, result} ->
+      case validate_option(ele, options) do
+        {:ok, value} ->
+          {:cont, {:ok, Keyword.put(result, key, value)}}
+
+        {:error, _} = result ->
+          {:halt, result}
+      end
+    end)
+  end
+
+  defp validate_option({key, spec}, options) do
+    value = Keyword.get(options, key, spec[:default])
+    validate_type(spec[:type], key, value)
+  end
+
+  defp validate_type(:boolean, key, value) when not is_boolean(value),
```

Comment:
`when not is_boolean(value)` this forces all the options to be of boolean type.

Can we simplify the implementation by just pattern matching on `:attribute_key` & `:attribute_value`?

Each option can have a different type of value, the module should support that.

Reply:
I am keeping the implementation specific to only possible control options and their corresponding data-types.